File: data/src/main/java/com/ykb/app/cryptotrader/data/dao/RequestUriDao.java
package com.ykb.app.cryptotrader.data.dao;

import com.ykb.app.cryptotrader.data.model.RequestUri;
import com.ykb.app.cryptotrader.data.repo.RequestUriRepo;
import org.springframework.http.HttpMethod;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public final class RequestUriDao extends Dao<RequestUri, RequestUri.Key> {

    public RequestUriDao(RequestUriRepo repo) {
        super(repo);
    }

    public List<RequestUri> findByUriAndMethodAndServiceName(String uri, HttpMethod httpMethod) {
        return ((RequestUriRepo) repo).findByUriAndMethodAndServiceName(uri, httpMethod.name());
    }

}

File: library/data/data-auth/src/main/kotlin/com/ykb/app/cryptotrader/data/auth/dao/RequestUriDao.kt
package com.ykb.app.cryptotrader.data.auth.dao

import com.ykb.app.cryptotrader.data.auth.model.RequestUri
import com.ykb.app.cryptotrader.data.auth.repo.RequestUriRepo
import com.ykb.app.cryptotrader.data.base.Dao
import org.springframework.http.HttpMethod
import org.springframework.stereotype.Service

@Service
class RequestUriDao(repo: RequestUriRepo) : Dao<RequestUri, RequestUri.Key>(repo) {

    fun findByServiceName(serviceName: String): List<RequestUri> {
        return (repo as RequestUriRepo).findByServiceName(serviceName)
    }

    fun findByUriAndMethodAndServiceName(uri: String, httpMethod: HttpMethod, serviceName: String): List<RequestUri> {
        return (repo as RequestUriRepo).findByUriAndMethodAndServiceName(uri, httpMethod.name(), serviceName)
    }

}